dev_docs/tutorials/saved_objects.mdx

```diff
@@ -161,134 +162,29 @@ identify this reference. This guarantees that the id the reference points to alw
-
-[2] Returning a shallow copy is necessary to avoid type errors when using different types for the input and output shape.
-
-[3] Migrations do not have to be defined for every version. The version number of a migration must always be the earliest Kibana version
- in which this migration was released. So if you are creating a migration which will
- be part of the v7.10.0 release, but will also be backported and released as v7.9.3, the migration version should be: 7.9.3.
-
-Migrations should be written defensively, an exception in a migration function will prevent a Kibana upgrade from succeeding and will cause downtime for our users. 
-Having said that, if a document is encountered that is not in the expected shape, migrations are encouraged to throw an exception to abort the upgrade. In most scenarios, it is better to
-fail an upgrade than to silently ignore a corrupt document which can cause unexpected behaviour at some future point in time. When such a scenario is encountered,
-the error should be verbose and informative so that the corrupt document can be corrected, if possible.
-
-**WARNING:** Do not attempt to change the `typeMigrationVersion`, `id`, or `type` fields within a migration function, this is not supported.
-
-### Deferred Migrations
-Usually, migrations run during the upgrade process, and sometimes that may block it if there is a huge amount of outdated objects.
-In this case, it is recommended to mark some of the migrations to defer their execution.
-
-```ts
-export const dashboardVisualization: SavedObjectsType = {
-  name: 'dashboard_visualization', [1]
-  /** ... */
-  migrations: {
-    // Takes a pre 1.1.0 doc, and converts it to 1.1.0
-    '1.1.0': {
-      deferred: true,
-      transform: migrateDashboardVisualization110,
-    },
-  },
-};
-```
-
-By default, all the migrations are not deferred, and in order to make them so, the `deferred` flag should be explicitly set to `true`.
-In this case, the documents with only pending deferred migrations will not be migrated during the upgrade process.
-
-But whenever they are accessed via Saved Object API or repository, all the migrations will be applied to them on the fly:
-- On read operations, the stored objects remain untouched and only transformed before returning the result.
-  If there are some failures during the migration, an exception or 500 server error will be thrown,
-  so that it is guaranteed that all the returned objects will be up to date.
-- On write operations, the objects will be migrated to the latest version before writing them.
-
-In other words, this flag postpones the write operation until the objects are explicitly modified.
-
-One important notice: if there is a few pending migrations for a document and not all of them can be deferred,
-the document will be migrated during the upgrade process, and all pending migrations will be applied.
+The first version must be numbered as version 1, incrementing by one for each new version.
```

Comment:
I think adding a code example after this line would be quite useful!

Reply:
we already have an example: https://github.com/elastic/kibana/pull/178058/files/7c3441ba4f9c2e26cd10a5fba570cac0100a4ba8#diff-7ba839dad2b02057116b203764e2f4b085f1014a745f499cbba73fff88524cbdR37. I'll add another one if you think it's useful.

Reply:
Done:
![Screenshot 2024-03-06 at 11 14 51](https://github.com/elastic/kibana/assets/11909450/a5c4c0c6-ca6f-4dc9-8b3f-5fd02835cecb)

Reply:
Good point! I was specifically thinking of fleshing out the `modelVersions` part with detail about how the migrations might look.

I've pushed `f778b86a765c590eb9b0118595c8dcf3e2610623` to this PR, let me know what you think! Feel free to edit or revert it!